File: 360MedLink-Back/src/main/java/org/sid/entities/Salary.java
package org.sid.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.time.LocalDate;
import java.util.Date;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
public class Salary {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NotNull
    private Integer iduser;

    @NotEmpty(message = "user titre id is required")
    private String titre;

    @NotEmpty(message = "salary is required")
    private String salary;

    private LocalDate date;




}

File: 360MedLink-Front/src/app/models/salary.ts
export class Salary {
  public id:number ;
  public iduser:number ;
  public titre:string ;
  public salary:string ;
  public date :string;

  constructor(id?: number, iduser?: number, titre?: string, salary?: string,date?:string) {
    this.id     = id=0;
    this.iduser = iduser=0;
    this.titre  = titre='';
    this.salary = salary='';
    this.date   = date='';
  }
}
